src/System.IO.Pipelines/System/IO/Pipelines/Pipe.cs

```diff
@@ -257,7 +257,8 @@ internal void Append(ReadableBuffer buffer)
 
             EnsureAlloc();
 
-            var clonedBegin = BufferSegment.Clone(buffer.Start, buffer.End, out BufferSegment clonedEnd);
+            // TODO: this has to support array buffers
+            var clonedBegin = BufferSegment.Clone(buffer.Start.GetSegment(), buffer.Start.Index, buffer.End.GetSegment(), buffer.End.Index, out BufferSegment clonedEnd);
```

Comment:
What happens if the underlying item is an array? Does it throw? Should we have a skipped test for this?

Reply:
It throws. Pipe doesn't support non linked list buffers and there are not tests for this.